### PR TITLE
Convert from Azure Pipelines to GitHub actions

### PR DESCRIPTION
**Pull request by thetorpedodog**
Pretty self-explanatory from the commits and the description, yeah?

[ch10037]

### CODE REVIEW

**Comment by Shelnutt2** [comment]
> Pretty self-explanatory from the commits and the description, yeah?
> 
> [ch10037]

bigger change is finally removing the dead pipfile! :tada:

File: .github/workflows/tiledb-cloud-py.yaml
Comment (Shelnutt2) [inline .github/workflows/tiledb-cloud-py.yaml:8]: Do you want to run on all pull requests instead of only master? That way if you do a PR into a different branch it runs?

Comment (thetorpedodog) [reply .github/workflows/tiledb-cloud-py.yaml:8]: Good idea. Changed.

**Comment by thetorpedodog** [comment]
This is, in theory, good to go. Unfortunately it looks a lot like the mysterious hangs are still happening, though now in a different place: after running everything in test_basic.py, and also only sometimes. This is an improvement at least?

My feelings are that, for now, we should drop the requirement that the whole test matrix passes so that we’re not completely blocked by test environment problems. We in the company will just have to run `pytest` locally in both py3.6 and py3.7 and note that we have done so in a comment. If the pipelines *do* finish, they will still give us some extra confidence. (@ihnorton, @nguyenv, @Shelnutt2: thoughts?)

I’m going to smash that “re-run jobs” button, eat lunch, and see if it has decided to behave once I get back.

**Comment by ihnorton** [comment]
3.6 and 3.7 are the only versions supported by UDFs, so that's not a viable solution IMO. I would suggest reverting #157 on a branch to see if the issues are caused by those changes vs potential issues w/ newly-released versions of dependencies.

**Comment by thetorpedodog** [comment]
figured it out. it was `test_tasks` the whole time. Azure Pipelines and the lack of output was just a red herring.

Added 2667c7352b0f67aee03a78ca2cb70eacfcccdb72 to the start of the change history and all the tests seem to be happily running.

On the plus side, getting off of Azure Pipelines and into the better-integrated GitHub Actions is still a net benefit.

**Comment by thetorpedodog** [comment]
last attempt failed due to things that were Not Our (in the narrow sense of “this library’s”) Fault:

```
E           HTTP response headers: HTTPHeaderDict({'Date': 'Wed, 01 Sep 2021 18:50:29 GMT', 'Content-Type': 'text/html', 'Content-Length': '150', 'Connection': 'keep-alive'})
E           HTTP response body: <html>
E           <head><title>502 Bad Gateway</title></head>
E           <body>
E           <center><h1>502 Bad Gateway</h1></center>
E           <hr><center>nginx</center>
E           </body>
E           </html>
```

**Comment by ihnorton** [comment]
> On the plus side, getting off of Azure Pipelines and into the better-integrated GitHub Actions is still a net benefit.

💯

**Comment by thetorpedodog** [comment]
Everything passes! This is ready to go.

**Comment by ihnorton** [comment]
What is the reason for the version-specific `requirements` files with so many pins? This is a customer-facing package, so ideally we should test against un-pinned versions because that's what new installs will  (usually) see, unless there is a known issue.

**Comment by ihnorton** [comment]
Alternatively, I suppose we could have another matrix entry without pins - trying to understand the goal.

**Comment by thetorpedodog** [comment]
> What is the reason for the version-specific `requirements` files with so many pins? This is a customer-facing package, so ideally we should test against un-pinned versions because that's what new installs will (usually) see, unless there is a known issue.

The `requirements-pyX.Y.txt` files were each generated by `pip freeze` so that we get a repeatable install if we use the requirements file (thus insulating us from test failures due to dependency changes). `pip install tiledb-cloud` (or anything which depends upon `tiledb-cloud`) will still fetch the latest versions of everything.

Added a matrix dimension for fresh install vs. pinned deps.

**Review by ihnorton — APPROVED** [review]
LGTM

**Comment by thetorpedodog** [comment]
Removed outdated “expected” status checks in the repository settings; I’ll add the new ones once they start appearing in search (presumably sometime after this merge).

**Comment by ihnorton** [comment]
One intermediate option as follow-up is to use the pins for PRs in general but add a nightly task which runs against (and possibly updates?) unpinned versions of everything on pypi. @nguyenv added that recently for tiledb-py to test against core ‘dev’ on a nightly basis. It is a nice balance of stability for PRs along with proactive warning when something breaks in a dep (https://github.com/TileDB-Inc/TileDB-Py/blob/dev/.github/workflows/daily-test-build.yml)

**Comment by thetorpedodog** [comment]
I like that idea a lot! Once I get out of the stack of tasks I have going for myself I will look into it. Thank you!

**Pull request merged**
